### DISCUSSION TITLE
Tips & strategies for interpreting Teleport debug traces

### DISCUSSION THREAD

**Discussion by WilliamLoy** [gravitational/teleport#8160] [Q&A]
Does anyone have tips for interpreting Teleport debug traces?

**Comment by programmerq** [comment]
When teleport is in debug mode, that simply means that it will output messages of every possible log severity to its logs. When troubleshooting, this is helpful since you get the most information possible from the logging.

With teleport the log severities can be thought of generally as follows:

`DEBUG` - minute details that may or may not container helpful information. May mostly be noise in a working system.
`INFO` - non-critical informational messages about normal things happening.
`WARN` - when a less than ideal situation occurs
`ERRO` - teleport encounters an unexpected situation where it is not able to complete an operation correctly.

The teleport logs also include a component field, which helps identify which general subsystem is emitting the log. This appears in square brackets after the severity. Examples include `[AUTH]`, `[DYNAMODB]`, `[AUTH:COMP]`, `[S3]` and many others. Some log messages do not include a component and leave that field blank instead.

The next item that appears in a log entry is a generic message. This could be a simple word or phrase. It could be a full multiline stacktrace. This is often helpful in determining what exactly happened. The stacktrace can be utilized to step through the code to see lots of details about what was happening when the stacktrace was encountered. Not all errors include a stacktrace.

The last section of the logs is the file and line number from where the log message was emitted. This is helpful for investigating messages that don't include a stacktrace.

Most of the time when debug mode is activated, there is some reason to extract as much information as possible about what is going on. The additional information helps give more clues to investigating the behavior that is in question.
